src/WebJobs.Script.Scaling/ArgumentUtils.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Azure.WebJobs.Script.Scaling
+{
+    [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", Justification = "By design")]
+    public static class ArgumentUtils
+    {
+        public static void ValidateArgument(string name, string arg)
+        {
+            if (string.IsNullOrEmpty(arg))
+            {
+                throw new ArgumentNullException(name, "Parameter must not be null or empty.");
+            }
+        }
+
+        public static void ValidateArgument(string name, object arg)
+        {
+            if (arg == null)
+            {
+                throw new ArgumentNullException(name, "Parameter must not be null.");
+            }
+        }
+
+        public static void ValidateArgument(IWorkerInfo worker)
+        {
+            ValidateArgument("worker", (object)worker);
+            ValidateArgument("worker.SiteName", worker.SiteName);
```

Comment:
Come to think of it, I will remove all this validation.   This is not intended for public (yet).  We will trust the runtime to pass a valid argument.   It is overkilled to second guess the runtime.